### PR TITLE
cf.data_vars skips coords

### LINKED ISSUE #177: cf.data_vars fails when there are coords with conflicting standard names
```python
from cf_xarray.datasets import popds
popds["ULAT"].attrs["standard_name"] = "latitude"
popds["TLAT"].attrs["standard_name"] = "latitude"
popds = popds.reset_coords("ULAT")
popds.cf.data_vars
```
Error:
```
KeyError: "Receive multiple variables for key 'latitude': {'ULAT', 'TLAT'}. Expected only one. Please pass a list ['latitude'] instead to get all variables matching 'latitude'."
```

This shouldn't happen as `ULAT` is the only latitude in `data_vars`.

### PR DESCRIPTION
**Pull request by malmans2**
Close #177

### CODE REVIEW

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -629,11 +629,16 @@ def check_results(names, k):
                 varnames.extend(measure)
         else:
             stdnames = set(_get_with_standard_name(obj, k))
-            check_results(stdnames, k)
-            successful[k] = bool(stdnames)
             objcoords = set(obj.coords)
-            varnames.extend(stdnames - objcoords)
-            coords.extend(stdnames & objcoords)
+            stdvars = stdnames - objcoords
```
Comment (dcherian) [inline cf_xarray/accessor.py:633]: move stdvars into the else block and add a comment?

File: cf_xarray/accessor.py [hunk 2]
```diff
@@ -630,15 +630,12 @@ def check_results(names, k):
...
-            if "coords" not in skip:
-                check_results(stdnames, k)
-                successful[k] = bool(stdnames)
-                coords.extend(stdnames & objcoords)
-            else:
-                check_results(stdvars, k)
-                successful[k] = bool(stdvars)
-            varnames.extend(stdvars)
+            if "coords" in skip:
+                stdnames -= objcoords
```
Comment (malmans2) [inline cf_xarray/accessor.py:634]: How about this, exactly the same as the origin with an extra if statement?

**Comment by malmans2** [comment]
I don't think the error in the doc notebook is related to this PR. Opening a separate issue.

File: cf_xarray/accessor.py (hunk 2 above)
Comment (dcherian) [reply cf_xarray/accessor.py:634]: Perfect.

**Comment by dcherian** [comment]
Looks like the demo notebook is now broken.

**Comment by dcherian** [comment]
ah just saw your comment.

**Pull request merged**